src/controllers/midi/midicontroller.cpp

```diff
@@ -494,8 +505,8 @@ void MidiController::processInputMapping(const MidiInputMapping& mapping,
         if (pEngine == NULL) {
             return;
         }
-        QScriptValue function = pEngine->wrapFunctionCode(mapping.control.item, 2);
-        if (!pEngine->execute(function, data, timestamp)) {
+        QJSValue function = pEngine->wrapFunctionCode(mapping.control.item, 2);
+        if (!pEngine->executeFunction(function, data)) {
```

Comment:
Why is the time stamp parameter omitted?

Reply:
I think this was an accident, good catch.

Reply:
The timestamp wasn't passed to the script before: https://github.com/mixxxdj/mixxx/pull/2682/files#diff-9289ba7fe799f555352e9653196f4b00L488

Reply:
(That link doesn't take me to a line.) The diff referenced shows that it was. Regardless, time stamps are needed for accurate tracking/scratching with moving-platter controllers.

Reply:
Sorry, the link above doesn't work after rebasing. Here are links to the code in 2.3:
https://github.com/mixxxdj/mixxx/blob/2.3/src/controllers/controllerengine.cpp#L479
https://github.com/mixxxdj/mixxx/blob/2.3/src/controllers/controllerengine.cpp#L495
The timestamps have never been passed to scripts. I will not make any further API changes to ControllerEngine. Instead, we will be working on a new controller system from scratch. Timestamps for scratching should probably be passed to audio thread via a lock free queue. The timing of message passing between ControllerEngine and the audio thread now is unpredictable using a ControlObject.